rskj-core/src/main/java/co/rsk/net/light/LightProcessor.java

```diff
@@ -0,0 +1,222 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.net.light;
+
+import co.rsk.core.RskAddress;
+import co.rsk.crypto.Keccak256;
+import co.rsk.db.RepositorySnapshot;
+import co.rsk.db.RepositoryLocator;
+import co.rsk.net.light.message.*;
+import org.ethereum.core.*;
+import org.ethereum.net.message.Message;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.db.BlockStore;
+import org.ethereum.db.TransactionInfo;
+import org.ethereum.vm.DataWord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.LinkedList;
+import java.util.List;
+
+
+/**
+ * Created by Julian Len and Sebastian Sicardi on 21/10/19.
+ */
+public class LightProcessor {
+    private static final Logger logger = LoggerFactory.getLogger("lightprocessor");
+    // keep tabs on which nodes know which blocks.
+    private final BlockStore blockStore;
+    private final RepositoryLocator repositoryLocator;
+    private final Blockchain blockchain;
+
+    public LightProcessor(@Nonnull final Blockchain blockchain,
+                          @Nonnull final BlockStore blockStore,
+                          @Nonnull final RepositoryLocator repositoryLocator) {
+        this.blockchain = blockchain;
+        this.blockStore = blockStore;
+        this.repositoryLocator = repositoryLocator;
+    }
+    /**
+     * processBlockReceiptsRequest sends the requested block receipts if it is available.
+     * @param requestId the id of the request
+     * @param blockHash   the requested block hash.
+     * @param lightPeer
+     */
+    public void processGetBlockReceiptsMessage(long requestId, byte[] blockHash, LightPeer lightPeer) {
+        String blockHashLog = Hex.toHexString(blockHash);
+        logger.trace("Processing block receipts request {} block {}", requestId, blockHashLog);
+        final Block block = blockStore.getBlockByHash(blockHash);
+
+        if (block == null) {
+            // Don't waste time sending an empty response.
+            return;
+        }
+
+        List<TransactionReceipt> receipts = new LinkedList<>();
+
+        for (Transaction tx :  block.getTransactionsList()) {
+            TransactionInfo txInfo = blockchain.getTransactionInfo(tx.getHash().getBytes());
+            receipts.add(txInfo.getReceipt());
+        }
+
+        Message responseMessage = new BlockReceiptsMessage(requestId, receipts);
+        lightPeer.sendMessage(responseMessage);
+    }
+
+    public void processBlockReceiptsMessage(long id, List<TransactionReceipt> blockReceipts, LightPeer lightPeer) {
```

Comment:
I would delete this and all the other methods that aren't supported. They don't belong together at all: One thing is to reply to LC queries as a full node, something very different is to consume replies to those queries as a light node. I would also rename the class to reflect this, maybe something like LightClientQueriesResolver (I didn't think this through).